cmd/zapi/cmd/post/post.go

```diff
@@ -0,0 +1,125 @@
+package post
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"io"
+	"os"
+	"sync/atomic"
+	"time"
+
+	"github.com/brimsec/zq/cmd/zapi/cmd"
+	"github.com/brimsec/zq/cmd/zapi/format"
+	"github.com/brimsec/zq/pkg/display"
+	"github.com/brimsec/zq/zqd/api"
+	"github.com/mccanne/charm"
+)
+
+var Post = &charm.Spec{
+	Name:  "post",
+	Usage: "post [options] path...",
+	Short: "post log file(s) to a space",
+	New:   NewPost,
+}
+
+func init() {
+	cmd.CLI.Add(Post)
+}
+
+type LogCommand struct {
+	*cmd.Command
+	force      bool
+	bytesRead  int64
+	bytesTotal int64
+	start      time.Time
+	done       bool
+}
+
+func NewPost(parent charm.Command, flags *flag.FlagSet) (charm.Command, error) {
+	c := &LogCommand{Command: parent.(*cmd.Command)}
+	flags.BoolVar(&c.force, "f", false, "create space if specified space does not exist")
+	return c, nil
+}
+
+// Run lists all spaces in the current zqd host or if a parameter
+// is provided (in glob style) lists the info about that space.
+func (c *LogCommand) Run(args []string) (err error) {
```

Comment:
This method is nearly identical to (*PcapCommand).Run. Maybe refactor?

Reply:
Looked into this. It's got enough subtle differences to be more trouble then its worth. I definitely think we'll look into merging the pcap and log commands in the future.